### PR TITLE
added risk model and risk/vuln/attack/campaign forms and tree command

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/models/orgs.py [hunk 1]
```diff
@@ -64,8 +64,39 @@ def getModelDefs(self):
...
+                ('ou:campaign', ('guid', {}), {
+                    'doc': 'Represents an orgs activity in persuit of a goal',
+                }),
             ),
             'forms': (
+                ('ou:goal', {}, (
+                    ('name', ('str', {}), {
+                        'doc': 'A terse name for the goal',
+                    }),
+                    ('type', ('str', {}), {
```
Comment (vEpiphyte) [inline synapse/models/orgs.py:79]: strip / lower ?

File: synapse/models/risk.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                    'doc': 'An instance of an actor attacking a target.'
+                }),
+            ),
+            'forms': (
+                ('risk:vuln', {}, (
+                    ('name', ('str', {}), {
+                        'doc': 'A user specified name for the vulnerability',
+                    }),
+
+                    ('type', ('str', {}), {
```
Comment (vEpiphyte) [inline synapse/models/risk.py:25]: strip / lower ?

File: synapse/models/risk.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+                    ('org', ('ou:org', {}), {
+                        'doc': 'The vulnerable org',
+                    }),
+                    ('host', ('it:host', {}), {
+                        'doc': 'The vulnerable host',
+                    }),
+                    ('place', ('geo:place', {}), {
+                        'doc': 'The vulnerable place',
+                    }),
+                    ('software', ('it:prod:softver', {}), {
```
Comment (vEpiphyte) [inline synapse/models/risk.py:52]: array?

File: synapse/models/risk.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+                    ('campaign', ('ou:campaign', {}), {
+                        'doc': 'Set if the attack was part of a larger campaign',
+                    }),
+                    ('actor:org', ('ou:org', {}), {
+                        'doc': 'The org that carried out the attack',
+                    }),
+                    ('actor:person', ('ps:person', {}), {
+                        'doc': 'The person that carried out the attack',
+                    }),
+                    ('target:org', ('ou:org', {}), {
```
Comment (vEpiphyte) [inline synapse/models/risk.py:82]: taget -> ndef ?
target:form breakout ?

Comment (vEpiphyte) [reply synapse/models/risk.py:82]: there is a disjoint between "things that are vulnerable" vs "things that are targeted"

File: synapse/models/risk.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+                        'doc': 'The target host was contacted via the IPv6 address',
+                    }),
+                    ('via:email', ('inet:email', {}), {
+                        'doc': 'The target person/org was contacted via the email address',
+                    }),
+                    ('via:phone', ('tel:phone', {}), {
+                        'doc': 'The target person/org was contacted via the phone number',
+                    }),
+                    ('used:vuln', ('risk:vuln', {}), {
+                        'doc': 'The actor used the vuln in the attack',
```
Comment (vEpiphyte) [inline synapse/models/risk.py:104]: chain of vulns?

File: synapse/models/orgs.py (hunk 1 above)
Comment (invisig0th) [reply synapse/models/orgs.py:82]: Decided against in case somebody wants to use them in titles/headers.

File: synapse/models/risk.py (hunk 4 above)
Comment (invisig0th) [reply synapse/models/risk.py:82]: there has to be.  software is vulnerable but a host is targeted.

File: synapse/models/risk.py [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+                        'doc': 'The vulnerable material item',
+                    }),
+                )),
+
+                ('risk:attack', {}, (
+                    ('time', ('time', {}), {
+                        'doc': 'Set if the time of the attack is known',
+                    }),
+                    ('success', ('bool', {}), {
+                        'doc': 'Set if the attack was known to have succeed or not',
```
Comment (Cisphyx) [inline synapse/models/risk.py:65]: Should be 'known to have succeeded' i think.

**Pull request merged**